server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -652,7 +653,9 @@ public List<? extends Host> discoverHosts(final AddHostCmd cmd) throws IllegalAr
             }
         }
 
-        return discoverHostsFull(dcId, podId, clusterId, clusterName, url, username, password, cmd.getHypervisor(), hostTags, cmd.getFullUrlParams(), false);
+        String hypervisorType = cmd.getHypervisor().equalsIgnoreCase(HypervisorGuru.HypervisorCustomDisplayName.value()) ?
+                "Custom" : cmd.getHypervisor();
```

Comment:
can there be only one type of custom hypervisor and hence only one `HypervisorCustomDisplayName`?

Reply:
Yes, only one